### PR TITLE
update motoko markdown docs to latest 

### PR DESCRIPTION
**Pull request by crusso**
Thank you for your contribution to the IC Developer Portal.
Before submitting your Pull Request, please make sure that:

- [ ] You have updated the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file according to your changes.
- [x] You have registered new documents to [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js) if any.

### CODE REVIEW

**Comment by crusso** [comment]
@wackyleo459, @hugolgst  I was trying to update the motoko markdown docs (to use the new interpreter code block attributes) but CI is hanging on the last few commits - any idea what might be wrong?

**Comment by crusso** [comment]
Ok, all systems go so ready for review!

File: docs/developer-docs/build/languages/motoko/about-this-guide.md
Comment (sesi200) [inline docs/developer-docs/build/languages/motoko/about-this-guide.md:49]:
```suggestion
For transparency into the principles that guide the engineering effort, the engineering organization has identified the following sets of core values and secondary values for driving the direction of the Motoko programming language.
```

File: docs/developer-docs/build/languages/motoko/control-flow.md
Comment (sesi200) [inline docs/developer-docs/build/languages/motoko/control-flow.md:78]:
```suggestion
Like many other high-level languages, Motoko lets you opt in to `null` values, tracking possible occurrences of `null` values using option types of the form `?T`. This is to both to encourage you to avoid using `null` values when possible, and to consider the possibility of `null` values when necessary.
```
Occurrence: https://en.wiktionary.org/wiki/occurence

File: docs/developer-docs/build/languages/motoko/overview.md
Comment (sesi200) [inline docs/developer-docs/build/languages/motoko/overview.md:17]:
```suggestion
-   Seamless integration of Internet Computer features
```
should we spell it out here as well?

**Review by sesi200 — CHANGES_REQUESTED** [review]
This looks like it's been a huge amount of work. Thank you very much for the cleanup! Just a few nits I'd like to fix before merging.

File: docs/developer-docs/build/languages/motoko/overview.md
Comment (crusso) [reply docs/developer-docs/build/languages/motoko/overview.md:17]: > This looks like it was a huge amount of work. Thank you very much! Just a few more nits I'd like to clean up before merging.

Thanks - I'll fix.

**Comment by crusso** [comment]
@hugolgst PTAL (Please take a look)

**Review by sesi200 — APPROVED** [review]
LGTM, thanks a lot @crusso

File: docs/developer-docs/build/languages/motoko/actor-classes.md
Comment (wackyleo459) [inline docs/developer-docs/build/languages/motoko/actor-classes.md:9]: It's really cool to see the interpreter working in conjunction with the adocs! Just curious, is this the expected layout output you want to see? for "Buckets.mo" text?
screenshot

**Review by wackyleo459 — APPROVED** [review]
Thank you, it's nice to see the interpreter working with the codeblocks this way!
Just left one comment (more of an observation) on layout for codeblock section title? Which is more cosmetic.

This must have taken a lot of time to go over!

**Comment by sesi200** [comment]
@crusso I would have merged this myself if you didn't explicitly call for Hugo's review. Since it is an improvement over the current state I don't see a reason to not merge it. Feel free to merge.

**Pull request merged**
